src/solvers/smt2_incremental/smt2_incremental_decision_procedure.cpp

```diff
@@ -215,7 +229,8 @@ decision_proceduret::resultt smt2_incremental_decision_proceduret::dec_solve()
 {
   ++number_of_solver_calls;
   solver_process->send(smt_check_sat_commandt{});
-  const smt_responset result = solver_process->receive_response();
+  const smt_responset result =
```

Comment:
You are sending the `smt_check_sat_commandt` command to the solver two times here, once in the line above and once in the new helper function.

Is that an error or by design?

Reply:
Ah, that is a mistake I made in the addition of the success messages. I will fix this.